dotCMS/src/integration-test/java/com/dotcms/publisher/bundle/business/BundleAPITest.java

```diff
@@ -0,0 +1,205 @@
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser));
+
+        bundleAPI.deleteBundleAndDependencies(bundleIdUser,newUser);
+        assertNull(bundleAPI.getBundleById(bundleIdUser));
+
+        bundleAPI.deleteBundleAndDependencies(bundleIdAdmin,newUser);
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin));
+    }
+
+    @Test(expected = NotFoundInDbException.class)
+    public void test_deleteBundleAndDependencies_bundleIdDoesNotExist() throws DotDataException {
+        bundleAPI.deleteBundleAndDependencies(UUIDGenerator.generateUuid(),adminUser);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void test_deleteBundleAndDependenciesOlderThan_futureDate() throws DotDataException {
+        bundleAPI.deleteBundleAndDependenciesOlderThan(DateUtil.addDate(new Date(),
+                Calendar.MONTH,1),adminUser);
+    }
+
+    @Test
+    public void test_deleteBundleAndDependenciesOlderThan_byAdmin() throws DotDataException {
+        final User newUser = new UserDataGen().nextPersisted();
+        final String bundleIdAdmin = insertPublishingBundle(adminUser.getUserId(),new Date());
+        final String bundleIdUser = insertPublishingBundle(newUser.getUserId(),new Date());
+        final String bundleIdAdmin_futureDate = insertPublishingBundle(adminUser.getUserId(),DateUtil.addDate(new Date(),
+                Calendar.MONTH,1));
+
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser));
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin_futureDate));
+
+        Set<String> bundlesDeleted = bundleAPI.deleteBundleAndDependenciesOlderThan(new Date(),adminUser);
+        assertTrue(bundlesDeleted.contains(bundleIdAdmin));
+        assertTrue(bundlesDeleted.contains(bundleIdUser));
+        assertFalse(bundlesDeleted.contains(bundleIdAdmin_futureDate));
+
+        assertNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNull(bundleAPI.getBundleById(bundleIdUser));
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin_futureDate));
+    }
+
+    @Test
+    public void test_deleteBundleAndDependenciesOlderThan_byLimitedUser() throws DotDataException {
+        final User newUser = new UserDataGen().nextPersisted();
+        final String bundleIdAdmin = insertPublishingBundle(adminUser.getUserId(),new Date());
+        final String bundleIdUser = insertPublishingBundle(newUser.getUserId(),new Date());
+        final String bundleIdUser_futureDate = insertPublishingBundle(newUser.getUserId(),DateUtil.addDate(new Date(),
+                Calendar.MONTH,1));
+
+        assertNotNull(bundleAPI.getBundleById(bundleIdAdmin));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser));
+        assertNotNull(bundleAPI.getBundleById(bundleIdUser_futureDate));
+
+        Set<String> bundlesDeleted = bundleAPI.deleteBundleAndDependenciesOlderThan(new Date(),newUser);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'bundlesDeleted' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)